tools/ci/docker/linux/Dockerfile

```diff
@@ -191,6 +191,14 @@ RUN cd renesas-tools/source/newlib && \
 RUN cd /tools/renesas-tools/build/gcc && \
   make && make install
 
+# Install Rust and targets supported from NuttX
+ENV CARGO_HOME=/tools/rust
+RUN mkdir -p $CARGO_HOME && \
+  curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y && \
+  $CARGO_HOME/bin/rustup target add thumbv6m-none-eabi && \
+  $CARGO_HOME/bin/rustup target add thumbv7m-none-eabi
+ENV PATH="$CARGO_HOME/bin:$PATH"
```

Comment:
Not needed this will not be set on the final image. If it is needed in the final image set it there.

Reply:
What exactly I need to do? Are you just talking about the "ENV" part or the complete installation incl. "RUN"?

Reply:
So if you look though this file, there are multiple images being built for example the one that you added the rust tools to is 
 
 ​FROM​ nuttx-toolchain-base AS nuttx-toolchain-renesas


This lets us then write COPY from <image> in the final image which is then was is actually used. This both allows for parallel building of the image as well as for a much smaller final image. 

My point here a out the env bit is that it will not actually be used in the final image. That will only apply to the nuttx-toolchain-renesas image.

Reply:
I think I got and that's most likely the reason, why PR https://github.com/apache/incubator-nuttx-apps/pull/1021 is failing constantly.

Is there solution to place the "RUN" part at the beginning in "FROM builder-base AS nuttx-tools" and than at the end use "COPY" to put it into the final docker image under "FROM ubuntu:20.04"?

Reply:
You should be able to basically do this just instead copying the /tools/rust and using the cargo env. 

https://github.com/apache/incubator-nuttx/blob/280792f15c8f9a4649b6803a6f13d141c869905d/tools/ci/docker/linux/Dockerfile#L258-L259

Reply:
Please take a look at the following PR: https://github.com/apache/incubator-nuttx/pull/5579
I assume, the further discussion can be shifted to that PR, too.